B9PartSwitch/UI/UIPartActionSubtypeSelector.cs

```diff
@@ -111,6 +111,21 @@ public override void Setup(UIPartActionWindow window, Part part, PartModule part
             subtypeTitleText.text = switcherModule.CurrentSubtype.title;
 
             subtypeButtons[currentButtonIndex].Activate();
+
+            // up to 7 subtypeButtons fit in the PAW without scrolling
+            if (subtypeButtons.Count > 7)
+            {
+                scrollMain.scrollSensitivity = subtypeButtons.Count;
+                // scrollMain.horizontalNormalizedPosition is the left edge of the viewport relative to the content.
+                // scrollMain.viewport.rect.width will be 200+(2/3) units after layout stuff is finished.  Sadly, it is 0 when this code runs so we can't make use of it here.
+                // Each button is effectively 28 units wide, meaning the viewport is 7+(1/6) buttons wide.
+                // If we scroll past the last button, the buttons will bounce back so that the last button is touching the viewport's right edge.
+                // That bounce could make a player lose their place in the button list (plus it just looks bad).  Therefore, we don't use 1.0 to set the viewport full right.
+                const float viewportWidthInButtons = 7+1/6f;
```

Comment:
Maybe we could put the width in a constant somewhere and derive the rest of the numbers from it?

Reply:
Is the 28 unit button width already accessible somewhere?  I couldn't find it, but maybe I'm just not looking in the right place.  Everything I tried is still 0 or 1 when this code runs.  Without that, it's two constants (viewportWidth and buttonWidth) that would be needed.  I could certainly do it that way.  Let me know what you think/prefer.

Reply:
I'm not aware of it being accessible anywhere.  I could go either way on this

Reply:
It's unavoidably awkward either way, but overall I think a single viewportWidthInButtons constant is preferable to splitting it into two (e.g., viewportWidth and buttonWidth).  We only ever need the ratio (viewportWidth / buttonWidth); there's currently no use for either part individually.  I think there's more risk of confusion for future contributors with viewportWidth and buttonWidth, whereas viewportWidthInButtons sounds like a special-purpose value that shouldn't be used for anything else.